addons/sale_loyalty/models/sale_order.py

```diff
@@ -1010,3 +1010,9 @@ def _try_apply_code(self, code):
                 return apply_result
             coupon = apply_result.get('coupon', self.env['loyalty.card'])
         return self._get_claimable_rewards(forced_coupons=coupon)
+
+    def _compute_amount_total_without_delivery(self):
+        self.ensure_one()
+        # Add back 'payment' rewards from the total without delivery, they should count towards the delivery price goal.
+        lines = self.order_line.filtered(lambda l: l.coupon_id)
+        return super()._compute_amount_total_without_delivery() - sum(lines.mapped('price_unit'))
```

Comment:
I guess this can go in `loyalty_delivery` module (and therefore there won't be a need to add the method in `sale`)